pkg/scrape/caribbeancom.go

```diff
@@ -0,0 +1,130 @@
+		e.ForEach(`script`, func(id int, e *colly.HTMLElement) {
+			if !strings.Contains(e.Text, "movie_seq") {
+				return
+			}
+			jsonData := e.Text[strings.Index(e.Text, "{") : len(e.Text)-3]
+			movSeq := gjson.Get(jsonData, "movie_seq").String()
+			if movSeq == "" {
+				return
+			}
+			sc.SiteID = movSeq
+			sc.SceneID = slugify.Slugify(sc.Site) + "-" + sc.SiteID
+		})
+
+		// Title
+		e.ForEach(`h1[itemprop=name]`, func(id int, e *colly.HTMLElement) {
+			sc.Title = strings.TrimSpace(strings.Replace(e.Text, "[VR] ", "", 1))
+		})
+
+		// Cover
+		coverURL := strings.Replace(strings.Replace(sc.HomepageURL, "eng/", "", 1), "index.html", "images/poster_en.jpg", 1)
+		if len(coverURL) > 0 {
+			sc.Covers = append(sc.Covers, coverURL)
+		}
+
+		// Filename  011421-001-carib-2160p.mp4
+		sc.Filenames = append(sc.Filenames, strings.Split(coverURL, "/")[4]+"-carib-2160p.mp4")
+
+		// Gallery
+		e.ForEach(`div.movie-gallery a.fancy-gallery`, func(id int, e *colly.HTMLElement) {
+			if strings.Compare(e.Attr(`data-is_sample`), "0") == 0 {
+				return
+			}
+			sc.Gallery = append(sc.Gallery, e.Request.AbsoluteURL(e.Attr("href")))
+		})
+
+		// Cast & Tags
+		e.ForEach(`div.movie-info a.spec__tag`, func(id int, e *colly.HTMLElement) {
+			if strings.Compare(e.Attr(`itemprop`), "actor") == 0 {
+				sc.Cast = append(sc.Cast, strings.TrimSpace(e.Text))
+			} else {
+				if (strings.Compare(e.Attr(`itemprop`), "genre") == 0) || (strings.Compare(e.Attr(`itemprop`), "url") == 0) {
+					sc.Tags = append(sc.Tags, strings.TrimSpace(e.Text))
+				}
+			}
+		})
+
+		// Release Date
+		e.ForEach(`div.movie-info span`, func(id int, e *colly.HTMLElement) {
+			if e.Attr(`itemprop`) == "uploadDate" {
+				sc.Released = strings.TrimSpace(strings.Replace(e.Text, "/", "-", -1))
+			}
+			// Duration
+			if e.Attr(`itemprop`) == "duration" {
+				tmpDuration := strings.Split(strings.Trim(e.Attr(`content`), "TS"), "M")[0]
+				sc.Duration, _ = strconv.Atoi(strings.Split(tmpDuration, "H")[1])
+			}
+		})
+
+		sceneURLJap := strings.Replace(strings.Replace(sc.HomepageURL, "eng/", "", 1), "en.", "www.", 1)
+		ctx := colly.NewContext()
+		ctx.Put("scene", sc)
+
+		sceneCollectorJap.Request("GET", sceneURLJap, nil, ctx, nil)
+	})
+
+	// Synopsis - Pull from Japanese site & translate
+	sceneCollectorJap.OnHTML(`html`, func(e *colly.HTMLElement) {
+		sc := e.Request.Ctx.GetAny("scene").(models.ScrapedScene)
+		e.ForEach(`p[itemprop=description]`, func(id int, e *colly.HTMLElement) {
+			sc.Synopsis, _ = gtranslate.Translate(strings.TrimSpace(e.Text), language.Japanese, language.English)
+		})
+
+		out <- sc
+	})
+
+	siteCollector.OnHTML(`div.media-thum a`, func(e *colly.HTMLElement) {
+		sceneURL := e.Request.AbsoluteURL(e.Attr("href"))
+		// If scene exists in database, there's no need to scrape
+		if !funk.ContainsString(knownScenes, sceneURL) {
+			sceneCollector.Visit(sceneURL)
+		}
+	})
+
+	siteCollector.Visit("https://en.caribbeancom.com/eng/listpages/vr1.htm")
```

Comment:
Just two more videos and we'll have to scrape more than one page, so let's future proof this already.

```suggestion
	siteCollector.OnHTML(`.pagination-large .pagination__item[rel="next"]`, func(e *colly.HTMLElement) {
		pageURL := e.Request.AbsoluteURL(e.Attr("href"))
		siteCollector.Visit(pageURL)
	})

	siteCollector.Visit("https://en.caribbeancom.com/eng/listpages/vr1.htm")
```

Reply:
change made. Thanks